CONTRIBUTING.md

```diff
@@ -182,3 +182,10 @@ public RealmQuery<E> equalTo(String fieldName, String fieldValue, boolean caseSe
 
 Above is based on the official guidelines from Oracle regarding Javadoc: http://www.oracle.com/technetwork/articles/java/index-137868.html
 
+### Branch Strategy
+
+* `releases` branch is where the hotfix versions released from. Releases from this branch will bump the last part of the version number.
+* `master` branch is where the major/minor versions released from. The first or the second part of the version number will be bumped when release from this branch.
+* Every commit to the `releases` branch will be merged to the master brunch automatically when it gets merged.
+* Only new features and breaking changes should go to the `master` branch.
+* Minor fixes like documentation fix, tests fix and build system changes should be committed to the `releases` branch to avoid diversity from `master`. Although those changes might not add any visible values to the end user.
```

Comment:
Although minor fixes to documentation, tests and build system might not be visible to end users, they should still be committed to `releases` to avoid divergence.